src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -207,6 +190,84 @@ func (r *StatefulSetResource) rollingUpdate(
 	return nil
 }
 
+func (r *StatefulSetResource) rollingRestart(ctx context.Context, pod, artificialPod *corev1.Pod, newVolumes map[string]interface{}) error {
+	opts := []patch.CalculateOption{
+		patch.IgnoreStatusFields(),
+		ignoreKubernetesTokenVolumeMounts(),
+		ignoreDefaultToleration(),
+		ignoreExistingVolumes(newVolumes),
+	}
+
+	patchResult, err := patch.DefaultPatchMaker.Calculate(pod, artificialPod, opts...)
+	if err != nil {
+		return err
+	}
+
+	if patchResult.IsEmpty() {
+		return nil
+	}
+
+	var ordinal int64
+	ordinal, err = strconv.ParseInt(pod.Name[len(r.pandaCluster.Name)+1:], 10, 32)
+	if err != nil {
+		return fmt.Errorf("cluster %s: cannot convert pod name (%s) to ordinal: %w", r.pandaCluster.Name, pod.Name, err)
+	}
+
+	if *r.pandaCluster.Spec.Replicas > 1 {
```

Comment:
is it possible for there to be a pod, but replicas being 0? 🤔 just curious

Reply:
This logic can be triggered only if you have at least 2 replicas. Redpanda reports:
```
WARN  2023-01-02 09:29:38,159 [shard 0] admin_api_server - admin_server.cc:392 - [_anonymous] exception intercepted - url: [http://schema-registry-0.schema-registry.kuttl-test-alive-bat.svc.cluster.local.:9644/v1/brokers/0/maintenance] http_return_status[400] reason - seastar::httpd::bad_request_exception (Maintenance mode may not be used on a single node cluster)
```

if our end two end tests are using replication factor of one.